react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({ email: null });
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange() {
+    this.setState({ email: event.target.value });
+  }
+
+  handleSubmit() {
+    event.preventDefault();
+    console.log('e-mail:', this.state);
+  }
+
+  render() {
+    return <form onSubmit={this.handleSubmit}>
+      <label>
+        Email:
+      <input type="text" email={this.state.value} onChange={this.handleChange} placeholder="Enter e-mail here"/>
```

Comment:
The `email` attribute should be `value` and it should be set to the value in state that is holding the data for the input, in this case `this.state.email`